lib/components/FileAttachment/index.jsx

```diff
@@ -0,0 +1,188 @@
+  const [fileProgresses, setFileProgresses] = useState(
+    attachments.map(attachment => ({
+      filename: attachment.filename,
+      signedId: attachment.signedId,
+      url: attachment.url,
+      progress: 100,
+    }))
+  );
+
+  const addFileRef = useRef(null);
+
+  const { uppy, isUploading } = useUppyUploader({
+    endpoint,
+    uppyConfig: DEFAULT_UPPY_CONFIG,
+    uppyUploadConfig: UPPY_UPLOAD_CONFIG,
+  });
+
+  const handleAddFile = event => {
+    const files = Array.from(event.target.files);
+    files.forEach(file => {
+      uppy.addFile({
+        name: file.name,
+        type: file.type,
+        data: file,
+      });
+    });
+
+    const newlyAddedFiles = [
+      ...uppy.getFiles().map(file => ({
+        filename: file.name,
+        signedId: "awaiting",
+        url: "",
+        progress: 0,
+      })),
+    ];
+    setFileProgresses(prevState => [...prevState, ...newlyAddedFiles]);
+    setAdding(true);
+  };
+
+  const handleUpload = async () => {
+    try {
+      const { successful = [] } = await uppy.upload();
+      const uploadedFiles = successful.map(file => ({
+        filename: file.name,
+        signedId: file.response.signed_id,
+        url: file.response.blob_url,
+      }));
+
+      setFileProgresses(prevState => [
+        ...prevState.filter(attachment => attachment.progress !== 100),
+      ]);
+      onChange([...attachments, ...uploadedFiles]);
+      setAdding(false);
+    } catch {}
+  };
+
+  const handleUploadProgress = (file, progress) => {
+    setFileProgresses(prevState =>
+      prevState.map(uploadingFile => ({
+        ...uploadingFile,
+        progress:
+          uploadingFile.filename !== file.name
+            ? uploadingFile.progress
+            : progress.progress,
+      }))
+    );
+  };
+
+  const handleDelete = async ({ signedId }) => {
+    try {
+      await directUploadsApi.destroy(endpoint, signedId);
+      onChange(
+        attachments.filter(attachment => attachment.signedId !== signedId)
+      );
+    } catch {}
+  };
+
+  const handleRename = async (attachment, callback) => {
+    try {
+      const { fileName, signedId } = attachment;
+      const payload = {
+        blob: { filename: fileName },
+      };
+
+      await directUploadsApi.update(endpoint, signedId, payload);
+      onChange(
+        attachments.map(attachment =>
+          attachment.signedId === signedId
+            ? {
+                ...attachment,
+                filename: fileName,
+              }
+            : attachment
+        )
+      );
+      callback(true);
+    } catch {}
+  };
+
+  const handleDownload = ({ url = "", filename = "file.txt" }) => {
+    saveAs(url, filename);
+  };
+
+  const DROPDOWN_OPTIONS = [
+    {
+      label: "Download",
+      handler: handleDownload,
+    },
+    {
+      label: "Rename",
+      handler: handleRename,
+    },
+    {
+      label: "Delete",
+      handler: handleDelete,
+    },
+  ];
```

Comment:
Can't we define all the `handleX` functions within the `Attachment.jsx` file? We need to define all functions within the closest referencee.